src/var/mutation_counting.jl

```diff
@@ -0,0 +1,152 @@
+# phylo/mutation_counting.jl
+# ==================
+#
+# Types and methods for counting different kinds of mutations between two
+# DNA or RNA sequences.
+#
+# Part of the Bio.Phylo module.
+#
+# This file is a part of BioJulia. License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+
+# Counting mutations
+# ------------------
+
+# Mutation types
+
+immutable Different end
+immutable Transition end
+immutable Transversion end
+
+
+"""
+    is_ambiguous_strict{T<:Nucleotide}(a::T, b::T)
+
+The strictest test of ambiguity, if either nucleotide is not
+A, T, G, or C, then this function returns true.
+"""
+@inline function is_ambiguous_strict{T<:Nucleotide}(a::T, b::T)
+    return isambiguous(a) || isambiguous(b)
+end
+
+typealias alphabets Union{DNAAlphabet{2}, DNAAlphabet{4}, RNAAlphabet{2}, RNAAlphabet{4}}
```

Comment:
All alphabet types are a subtype of `Bio.Seq.Alphabet`. Why do we need this alias?